torch/utils/data/dataset.py

```diff
@@ -253,6 +254,61 @@ def __len__(self):
         return total
 
 
+class BufferedShuffleDataset(IterableDataset[T_co]):
+    r"""Dataset shuffled from the original dataset.
+
+    This class is useful to shuffle an existing instance of an IterableDataset.
+    The buffer with buffer_size is filled with the items from the dataset first. Then,
+    each item will be yielded from the buffer by reservoir sampling via iterator.
+
+    For `buffer_size` smaller than 1, the `buffer_size` is set to 1 representing
+    non-shuffled dataset. For fully shuffling the whole dataset, the buffer size is
+    required to be greater than or equal to the size of dataset.
+
+    When it is used with :class:`~torch.utils.data.DataLoader`, each item in the
+    dataset will be yielded from the :class:`~torch.utils.data.DataLoader` iterator.
+
+    When it is used with :class:`~torch.utils.data.DataLoader` with :attr:`num_workers == 0`,
```

Comment:
is this meant to be an example of what to do?  This is a little grammatically strange, I expect a word before "manually"